app/services/interface_merge.rb

```diff
@@ -0,0 +1,23 @@
+class InterfaceMerge
+  def run(interfaces, compute_attrs)
+    vm_interfaces = compute_attrs.vm_interfaces
+
+    # merge with existing
+    interfaces.each do |nic|
+      if !nic.virtual?
+        vm_nic = vm_interfaces.shift
+        return if vm_nic.nil?
+        nic.compute_attributes = vm_nic
```

Comment:
```ruby
nic.compute_attributes = vm_nic
nic.compute_attributes["from_profile"] = compute_attrs.compute_profile.name
```
these lines could be extracted to some private `sync(nic, vm_nic, profile_name)` method so you don't repeat it later when creating new interfaces below

compute_attrs.compute_profile.name could be extracted to profile_name var

also use single quoted strings if there's no interpolation